Backends/CLX/mirror.lisp

```diff
@@ -122,6 +122,14 @@
     (xlib:display-force-output (clx-port-display port))))
 
 (defmethod port-shrink-sheet ((port clx-basic-port) (sheet mirrored-sheet-mixin))
-  (when-let ((mirror (sheet-direct-mirror sheet)))
-    (xlib:iconify-window (window mirror) (clx-port-screen port))
-    (xlib:display-force-output (clx-port-display port))))
+  ;; Not all x11 window-managers iconify windows (in particular many
```

Comment:
this method change illustrates a certain problematic pattern with the clim spec where user interactions and port interactions are somewhat contrived (I'll use shrink frame as an example, but this applies i.e to the window configuration event and resize-sheet too):

User: shrink-frame -> shrink-sheet -> port-shrink-sheet
Port: (handle-event iconify-event) -> shrink-sheet -> port-shrink-sheet

Using both leads to a situation where we may end in an infinite loop, because calling port-shrink-sheet could generate iconify-event that will lead to calling port-shrink-sheet again etc etc. In order to avoid such infinite loop CLIM-TOS has a keyword argument in a few functions names `port-did-it` that is `T` when the port generates the event and `NIL` when it is a result of the user interaction to distinguish both situations. McCLIM on the other hand (in the case of the configuration event) binds a dynamic variable `*configuration-event-p*` that inhibits calling `port-set-mirror-geometry` (in order to avoid the infinite loop). I wonder what uniform approach to this problem we should adopt.

Regarding this case, as noted it is wasteful to go through get-property twice in case when it is the port what leads to this call (via handle-event).

Reply:
Yes, for the moment I did not introduce any of the suggested approach (anyway there isn't infinite loop in this case), and I don't have opinion about what is better. 

I also don't understand exactly what is the right things to do in main gf (shrink-frame) and what to do in note-...... .

I also tried to check at the end of shrink-frame if the application is really shruk but it no so easy because the x11 mesages are intercepted from another thread (the port thread).

Reply:
OK, let's leave this problem to the future selves.